x-pack/test/alerting_api_integration/security_and_spaces/tests/actions/builtin_action_types/webhook.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { URL, format as formatUrl } from 'url';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import {
+  getExternalServiceSimulatorPath,
+  ExternalServiceSimulator,
+} from '../../../../common/fixtures/plugins/actions';
+
+const defaultValues: Record<string, any> = {
+  headers: null,
+  method: 'post',
+};
+
+function parsePort(url: Record<string, string>): Record<string, string | null | number> {
+  return {
+    ...url,
+    port: url.port ? parseInt(url.port, 10) : url.port,
+  };
+}
+
+// eslint-disable-next-line import/no-default-export
+export default function webhookTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  async function createWebhookAction(
+    urlWithCreds: string,
+    config: Record<string, string | Record<string, string>> = {}
+  ): Promise<string> {
+    const { url: fullUrl, user, password } = extractCredentialsFromUrl(urlWithCreds);
+    const url = config.url && typeof config.url === 'object' ? parsePort(config.url) : fullUrl;
+    const composedConfig = {
+      headers: {
+        'Content-Type': 'text/plain',
+      },
+      ...config,
+      url,
+    };
+
+    const { body: createdAction } = await supertest
+      .post('/api/action')
+      .set('kbn-xsrf', 'test')
+      .send({
+        description: 'A generic Webhook action',
+        actionTypeId: '.webhook',
+        secrets: {
+          user,
+          password,
+        },
+        config: composedConfig,
+      })
+      .expect(200);
+
+    return createdAction.id;
+  }
+
+  describe('webhook action', () => {
+    let webhookSimulatorURL: string = '<could not determine kibana url>';
+
+    // need to wait for kibanaServer to settle ...
+    before(() => {
+      const kibanaServer = getService('kibanaServer');
```

Comment:
Is there a missing es_archiver load here?

Reply:
It's on line 30.
Any reason to do it every time?
I haven't dug in to check for state issues, but I followed the example of the Slack tests.

Reply:
Ah cool, just missed it.